### PR TITLE
Upgrade Effect and define source adapter architecture

### PR DESCRIPTION
**Pull request by bmvantunes**
## What changed

- upgrade the workspace and Effect submodule to Effect 4.0.0-beta.100
- update protocol schema/codecs and package metadata for the beta.100 API
- document the unified Source Adapter architecture for materialized and leased topics
- define the browser-contract/runtime-composition boundary and standard adapter exports
- establish the canonical topic row identity and topic-owned source-module decisions
- keep planned adapter export examples from being interpreted as currently supported package imports

## Why

The runtime needs one strongly typed `source` seam that preserves the existing concise topic configuration while allowing first-party and third-party transports to supply materialized or leased streams. Browser consumers should remain unaware of Kafka, gRPC, or future adapter implementations.

Effect beta.100 is the baseline used to validate the proposed Effect services, Layers, Streams, scoped resource ownership, schemas, and typed-error design.

## Impact

This PR establishes the implementation contract and Effect dependency baseline. It does not implement the adapter runtime yet. Future adapter work can be divided into focused GitHub issues without redesigning the public API during implementation.

## Validation

- `vp install`
- focused package-seam validation
- repository policy tests: 362 passing with 100% coverage
- Effect diagnostics: zero errors, warnings, or messages
- package runtime and type tests, including protocol, client, config, engine, facade, and browser examples
- formatting, linting, and type checking passed during the validation runs

The aggregate `vp run -w ready` command ultimately encountered a `tsgolint` Go parser panic while linting the generated `examples/kafka-react/src/routeTree.gen.ts`. The substantive checks above passed; the remaining failure is a toolchain parser crash on generated TanStack route-tree code rather than a reported source diagnostic or failing assertion.

## Summary by CodeRabbit

- **New Features**
  - Added a canonical optional `source` configuration model for topics, plus a shared frozen View Server Config.
  - Standardized Source Adapter contracts for materialized and leased ingestion.
  - Introduced a canonical `id` for every Topic Row.
- **Bug Fixes**
  - Hardened live-wire JSON encoding/decoding and query/row envelope validation, with improved failures for malformed data and unexpected accessor properties.
  - Increased decoding safety via staged materialization and plan-based encode/decode.
- **Documentation**
  - Expanded ingestion architecture and added multiple ADRs covering adapter packaging/export seams and topic ownership.
- **Tests**
  - Strengthened hostile-input and boundary-condition coverage across wire codecs.
- **Chores**
  - Updated Effect dependencies and refreshed the Effect submodule reference.

### CODE REVIEW

File: CONTEXT.md [hunk 1]
```diff
@@ -422,8 +609,195 @@ _Avoid_: Browser write, send, emit
...
+- Serializable adapter option subtrees may use Effect Schema or Effect Config, while executable codecs, Mapping, Row Key functions, Schedules, Effects, and service references use exact TypeScript contracts plus adapter-owned construction validation.
+- The Source Adapter SDK validates and snapshots its common Source Definition envelope, and adapter-specific option validation completes before the Source Adapter Runtime Service starts it.
+- Every **Source Adapter Package Surface** exposes a browser-safe `/contract`, a server-only `/server`, and optional platform-specific Layer exports such as `/node`.
+- View Server exposes the Source Adapter SDK only through `effect-view-server/source-adapter`, `effect-view-server/source-adapter/server`, and `effect-view-server/source-adapter/testing`; package export checks reject deep or internal SDK imports.
+- First-party Kafka and gRPC Source Adapters are ordinary SDK consumers exposed through `effect-view-server/kafka/contract`, `effect-view-server/kafka/server`, `effect-view-server/kafka/node`, `effect-view-server/grpc/contract`, `effect-view-server/grpc/server`, and `effect-view-server/grpc/node`.
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply CONTEXT.md]: Fixed in 52465fe. CONTEXT and ADR 0009 now explicitly describe these as planned package surfaces whose implementation and export verification are pending; they are no longer presented as currently supported imports.

File: docs/adr/0006-one-canonical-topic-source-property.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Context
+
+Transport-specific `kafkaSource` and `grpcSource` properties keep ownership visible, but force configuration, public types, validation, Source Ownership Policy, and runtime composition to know every source kind. That prevents a third-party Source Adapter from joining the same source model without changing View Server itself.
+
+Splitting one application into a browser-safe topic contract and a second server-only topic tree would keep dependencies separate, but would make users describe the same View Server topology twice. That duplication is more error-prone and materially worse than the existing single-config API.
+
+## Decision
+
+A View Server Topic declares zero or one canonical `source` property in the one shared `defineViewServerConfig(...)` call. The value is an SDK-branded Source Definition created by a Source Adapter's browser-safe materialized or leased constructor. Arbitrary structural lookalikes are invalid. `kafkaSource` and `grpcSource` are removed in a hard migration with no aliases or compatibility layer.
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply docs/adr/0006-one-canonical-topic-source-property.md:15]: Fixed in 52465fe by explicitly deferring the migration. ADRs 0001, 0006, and 0010 now distinguish the current transport-specific source properties/configurable Topic keys from the accepted-but-unimplemented canonical `source`/`id` target architecture, so current public types and tests remain accurate.

File: packages/protocol/src/protocol-event-schema.ts [hunk 3]
```diff
@@ -8,6 +8,10 @@ export const ViewServerWireRowSchema: Schema.Codec<Schema.JsonObject> = Schema.R
 
 export type ViewServerWireRow = typeof ViewServerWireRowSchema.Type;
 
+export const ViewServerLooseWireRowSchema = Schema.Record(Schema.String, Schema.Unknown);
+
+export type ViewServerLooseWireRow = typeof ViewServerLooseWireRowSchema.Type;
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply packages/protocol/src/protocol-event-schema.ts]: Fixed in 52465fe. The exported loose row/event schemas were removed; the sole public wire schema remains JSON-shaped. Direct hostile input is now strict-materialized before any Schema traversal, with tests proving raw bigint, raw BigDecimal, outer accessors, and row accessors fail as typed `InvalidRow` without executing getters. Existing encoded bigint/BigDecimal round-trip coverage remains green.

File: docs/adr/0006-one-canonical-topic-source-property.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmvantunes) [reply docs/adr/0006-one-canonical-topic-source-property.md:15]: Fixed in 52465fe. CONTEXT now has a global current-vs-target status statement, ADRs 0001 and 0006 explicitly mark the canonical Source Adapter API as accepted target architecture with implementation pending, and all planned package paths/runtime composition are future-tense.

File: docs/adr/0010-canonical-topic-row-id.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Every Topic Row has one canonical string ID
+
+Every user-provided Topic Schema must declare the required field `id: Schema.String`, and Topic configuration has no separate `key` property. View Server rejects a missing, optional, transformed, refined, branded, or non-string `id` at compile time and defensively at runtime. This deliberately gives storage, queries, mutations, React rows, Wire Protocol schemas, and Source Adapters one universal identity contract. The common Source Adapter SDK requires only complete Upserts and ID-addressed Deletes; each adapter owns its ergonomic ID-producing API. Kafka composes `region:localRowKey`, while an adapter without that transport concept may require its Mapping to return the complete Topic Row directly.
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply docs/adr/0010-canonical-topic-row-id.md:7]: Fixed in 52465fe. ADRs 0001 and 0010 now explicitly state that canonical `id` and removal of the configurable `key` are accepted target architecture whose implementation is pending; the currently available configurable-key API is named as current behavior.

File: packages/protocol/src/protocol-event-codec.ts [hunk 5]
```diff
@@ -232,9 +233,12 @@ const decodeLiveEventWithContract = Effect.fn("ViewServerProtocol.event.decode")
...
   event: ViewServerWireEvent,
 ) {
-  const wireEvent = yield* Schema.decodeUnknownEffect(ViewServerWireEventSchema)(event).pipe(
-    Effect.mapError((error) => invalidRow(expectedTopic, `Invalid event: ${error.message}`)),
+  const materializedEvent = yield* materializeJsonFieldValue(event, (message) =>
+    invalidRow(expectedTopic, `Invalid event: ${message}`),
   );
+  const wireEvent = yield* Schema.decodeUnknownEffect(ViewServerWireEventSchema)(
+    materializedEvent,
+  ).pipe(Effect.mapError((error) => invalidRow(expectedTopic, `Invalid event: ${error.message}`)));
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply packages/protocol/src/protocol-event-codec.ts:196]: Fixed in 636c156. Snapshot and insert/update delta decoding now use internal already-materialized raw/grouped row decoders after event admission, eliminating the second recursive row traversal/allocation. Standalone row decoders retain strict materialization for hostile callers, and the trusted RPC path also benefits. Protocol remains 81/81 with 100% coverage; both 1,000-row × 10-subscriber WebSocket benchmark cases pass repository latency, backpressure, and cleanup thresholds.

**Comment by bmvantunes** [comment]
@codex review

File: packages/protocol/src/protocol-row-codec.ts [hunk 6]
```diff
@@ -198,26 +200,24 @@ export const encodeProjectedRow = Effect.fn("ViewServerProtocol.row.project.enco
...
   topic: Topic,
   selectedFields: ReadonlySet<string>,
   row: ViewServerWireRow,
 ) {
   const output: Record<string, unknown> = {};
-  const materialized = yield* materializeWireRow(topic, "row", row);
   for (const field of selectedFields) {
     const fieldSchema = config.topics[topic]!.schema.fields[field]!;
-    if (!Object.hasOwn(materialized, field) && !rowFieldIsOptional(fieldSchema)) {
+    if (!Object.hasOwn(row, field) && !rowFieldIsOptional(fieldSchema)) {
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply packages/protocol/src/protocol-row-codec.ts:395]: Fixed in 715e3f6. Every selected/grouped field now resolves through an own-property schema lookup before optionality, AST, encode, or decode work. Unknown and inherited field names fail with typed InvalidRow errors instead of defecting. Coverage includes raw selected fields, grouped fields, aggregate source fields, and hostile prototype/reflection cases across direct and live-event paths.

File: packages/protocol/src/protocol-row-codec.ts [hunk 7]
```diff
@@ -198,26 +200,24 @@ export const encodeProjectedRow = Effect.fn("ViewServerProtocol.row.project.enco
...
->(
+export const decodeMaterializedProjectedRow = Effect.fn(
+  "ViewServerProtocol.row.project.decodeMaterialized",
+)(function* <const Topics extends TopicDefinitions, Topic extends Extract<keyof Topics, string>>(
   config: { readonly topics: Topics },
   topic: Topic,
   selectedFields: ReadonlySet<string>,
   row: ViewServerWireRow,
 ) {
   const output: Record<string, unknown> = {};
```
[the remark this answers is not in the document]
Comment (bmvantunes) [reply packages/protocol/src/protocol-row-codec.ts:387]: Fixed in 715e3f6. The already-materialized raw/grouped decoders are now file-private and sit behind a compiled ViewServerEventRowPlan<Row>. Public codecs no longer accept a caller-selected Row generic: they derive LiveQueryRow from the configured topic schema plus exact query, validate the decoded exact shape and schema/aggregate value contract at runtime, and type encode/decode symmetrically. The server uses a separate honest object-typed runtime encoder. Type tests reject fabricated row claims and cover raw/grouped snapshot plus insert/update delta inference. The trusted hot path still performs no second materialization; both 1,000-row x 10-subscriber WebSocket baseline cases pass.

**Comment by bmvantunes** [comment]
@codex review

@coderabbitai review
